lib/tunnel.js

```diff
@@ -857,5 +877,30 @@ function clearTimeout_(timeoutId) {
     clearTimeout(timeoutId);
   }
 }
+/**
+ * killProcess_ is used to kill process method.
+ * @param {Object} that is Instance.
+ * @param {Function} fnCallback is Callback Method.
+ * @return {null} Return null.
+ */
+function killProcess_(that, fnCallback) {
+  that.proc.on('exit', function() {
+    if (that.proc.pid) {
+      setTimeout(function() {
+        try {
+          process.kill(that.proc.pid);
+        } catch (err) {}
+        return fnCallback();
+      }, 500);
+    }
+    return fnCallback();
+  });
+  if (os.platform() == 'win32') {
```

Comment:
Why is this condition in addition to an API call?
These should be used only when exit call sent non 200 response.

Reply:
Yes sir i did the same please see https://github.com/LambdaTest/node-tunnel/pull/5/files/447ed29bfbf7712c986a19b923db21cb83467de6#diff-7cb25e6733372586a7d0530feba6f8aeR487